htmresearch/algorithms/column_pooler.py

```diff
@@ -40,41 +34,39 @@ class ColumnPooler(object):
 
   def __init__(self,
                inputWidth,
-               lateralInputWidth,
-               numActiveColumnsPerInhArea=40,
+               lateralInputWidths=(),
+               cellCount=4096,
+               sdrSize=40,
+
+               # Proximal
                synPermProximalInc=0.1,
                synPermProximalDec=0.001,
                initialProximalPermanence=0.6,
-               columnDimensions=(2048,),
+               sampleSizeProximal=20,
                minThresholdProximal=10,
-               activationThresholdDistal=13,
-               minThresholdDistal=10,
-               initialPermanence=0.41,
-               connectedPermanence=0.50,
-               maxNewProximalSynapseCount=20,
-               maxNewDistalSynapseCount=20,
-               permanenceIncrement=0.10,
-               permanenceDecrement=0.10,
-               predictedSegmentDecrement=0.0,
-               maxSegmentsPerCell=255,
-               maxSynapsesPerProximalSegment=255,
-               maxSynapsesPerDistalSegment=255,
+               connectedPermanenceProximal=0.50,
+
+               # Distal
+               synPermDistalInc=0.1,
+               synPermDistalDec=0.001,
+               initialDistalPermanence=0.6,
+               sampleSizeDistal=20,
+               minThresholdDistal=13,
+               connectedPermanenceDistal=0.50,
+               distalSegmentInhibitionFactor=1.5,
+
                seed=42):
     """
-    This classes uses an ExtendedTemporalMemory internally to keep track of
-    distal segments. Please see ExtendedTemporalMemory for descriptions of
-    constructor parameters not defined below.
-
     Parameters:
     ----------------------------
     @param  inputWidth (int)
             The number of proximal inputs into this layer
 
-    @param  lateralInputWidth (int)
-            The number of lateral inputs into this layer
+    @param  lateralInputWidths (list of ints)
+            The number of input bits in each lateral input layer
```

Comment:
Maybe this is better:

~~~python
"""
    @param  inputWidth (int)
            The number of bits in the feedforward input

    @param  lateralInputWidths (list of ints)
            The number of bits in each lateral input
"""
~~~

I didn't see a reason to impose this limitation. We're free to send in input bits from any source, not necessarily just ColumnPoolers, and not necessarily ColumnPoolers of the same size.